frappe/email/receive.py

```diff
@@ -321,6 +328,46 @@ def process_part(self, part):
 		if part.get_filename():
 			self.get_attachment(part, charset)
 
+	def get_attached_message(self,part):
+		list=[None]*4
+		list[0]=part.__getitem__('from')
+		list[1]=part.__getitem__('to')
+		list[2]=part.__getitem__('subject')
+		list[3]=part.__getitem__('Date')
+
+		if None in list[:3]:#check if date returned if it didnt find a recieved field and get date from there
+			return False
```

Comment:
i think the return false could be changed to 
`if part.is_multipart():
     self.get_attached_message(part._payload[0])
return`
to remove the need for the last variable